test/jit/test_list_dict.py

```diff
@@ -1097,6 +1097,19 @@ def to_list_float_1D(x):
         self.checkScript(to_list_float_1D, (torch.randn(
             5, dtype=torch.double).cuda(),))
 
+    def test_no_element_type_annotation(self):
+        def fn(x):
+            # type: (torch.Tensor) -> List
+            a: List = x.tolist()
+            return a
+
+        with self.assertRaisesRegex(RuntimeError, r"Unknown type name"):
```

Comment:
self.checkScriptRaisesRegex is a useful utility for doing this; it will check eager/source FE/ast FE and compare results.

Reply:
No exception is raised in eager though, and there is no way to opt out of asserting that eager raises the exception too.